src/app/components/vending-machine/modal/_purchase-modal/AppPurchaseShopProductModal.vue

```diff
@@ -170,6 +172,54 @@ const balanceRefs = { coinBalance, joltbuxBalance };
 
 const processingPurchaseCurrencyId = ref<string>();
 
+let timerBuilder: NodeJS.Timer | null = null;
+const timeRemaining = ref<string>();
+
+function getTimeRemaining(endsOn: number) {
+	if (endsOn - getCurrentServerTime() < 1_000) {
+		if (timerBuilder) {
+			clearInterval(timerBuilder!);
+			timerBuilder = null;
+		}
+		timeRemaining.value = $gettext(`No longer for sale.`);
+		return;
+	}
+
+	timeRemaining.value = $gettextInterpolate(`Limited time to purchase: %{ time }`, {
+		time: shorthandReadableTime(endsOn, {
+			nowText: '0s',
+			precision: 'exact',
+			allowFuture: true,
+		}),
+	});
+}
+
+watch(
```

Comment:
Have you tried making this a watchEffect instead? Might simplify the code a bit.